app/ante/liquidate_spam_filter.go

```diff
@@ -0,0 +1,61 @@
+package ante
+
+import (
+	"fmt"
+
+	hardtypes "github.com/kava-labs/kava/x/hard/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+var _ sdk.AnteDecorator = HardLiquidateSpamFilter{}
+
+// HardLiquidateSpamFIlter blocks hard liquidate messages over a gas limit from the mempool
+type HardLiquidateSpamFilter struct {
+	gasLimit uint64
+}
+
+// NewHardLiquidateSpamFIlter creates a hard liquidate spam filter to filter liquidate messages with inflated gas
+func NewHardLiquidateSpamFilter(gasLimit uint64) HardLiquidateSpamFilter {
+	return HardLiquidateSpamFilter{
+		gasLimit: gasLimit,
+	}
+}
+
+// AnteHandle implements sdk.AnteDecorator
+func (f HardLiquidateSpamFilter) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	// only run in checktx to filter mempool
+	if !ctx.IsCheckTx() {
+		return next(ctx, tx, simulate)
+	}
+
+	// only apply to fee transactions
+	feeTx, ok := tx.(sdk.FeeTx)
+	if !ok {
+		return next(ctx, tx, simulate)
+	}
+
+	// check all tx messages for a hard liquidate
+	hasHardLiquidate := false
+	for _, msg := range tx.GetMsgs() {
+		if sdk.MsgTypeURL(msg) == sdk.MsgTypeURL(&hardtypes.MsgLiquidate{}) {
+			hasHardLiquidate = true
+			// one liquidate msg enough to end search
+			break
+		}
+	}
+
+	// continue if no hard liquidate messages are found
+	if !hasHardLiquidate {
+		return next(ctx, tx, simulate)
+	}
+
+	// reject transaction if greater than the set gas limit
+	if feeTx.GetGas() > f.gasLimit {
+		return ctx, sdkerrors.Wrapf(sdkerrors.ErrUnauthorized, "%v", fmt.Errorf("liquidation gas %d over %d limit", feeTx.GetGas(), f.gasLimit))
+	}
```

Comment:
i'd add a check that `next`s txs that have `fee < gasLimit` before we iterate through all the messages of the tx. 

that way we dont need to read into the messages for all low fee txs that will pass regardless of whether they have har liquidate messages or not.

Reply:
current logic:
* filter for hard liquidate message tx
* reject if gas > gas limit

proposed logic:
* continue if gas <= gas limit
* filter for hard liquidate messages & reject if found

Reply:
hmm seems like a good optimization will add